flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/resources/KubernetesPodsWatcherTest.java

```diff
@@ -65,7 +69,13 @@ void testClosingWithException() {
     void testCallbackHandler() {
         FlinkPod pod = new FlinkPod.Builder().build();
         final KubernetesPodsWatcher podsWatcher =
-                new KubernetesPodsWatcher(new TestingCallbackHandler(e -> {}));
+                new KubernetesPodsWatcher(
+                        TestingWatchCallbackHandler.<KubernetesPod>builder()
+                                .setOnAddedConsumer(pods -> podAddedList.addAll(pods))
+                                .setOnModifiedConsumer(pods -> podModifiedList.addAll(pods))
+                                .setOnDeletedConsumer(pods -> podDeletedList.addAll(pods))
+                                .setOnErrorConsumer(pods -> podErrorList.addAll(pods))
```

Comment:
It seems that these 4 list is only used in this test case. I'd suggestion moving their declaration to this method.